File: CS/CustomContextMenu/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Drawing;
using DevExpress.Utils;

namespace CustomContextMenu {
    public partial class Form1 : Form {
        const string aptDataFileName = @"..\..\Data\appointments.xml";
        const string resDataFileName = @"..\..\Data\resources.xml";
        
        public Form1() {
            InitializeComponent();
            schedulerControl1.Start = new DateTime(2010, 7, 05);
            schedulerControl1.DayView.TopRowTime = new TimeSpan(16, 0, 0);
            FillData();
        }

        #region FillData
        void FillData() {
            AppointmentCustomFieldMapping customNameMapping = new AppointmentCustomFieldMapping("CustomName", "CustomName");
            AppointmentCustomFieldMapping customStatusMapping = new AppointmentCustomFieldMapping("CustomStatus", "CustomStatus");
            schedulerStorage1.Appointments.CustomFieldMappings.Add(customNameMapping);
            schedulerStorage1.Appointments.CustomFieldMappings.Add(customStatusMapping);
            FillResourceStorage(schedulerStorage1.Resources.Items, resDataFileName);
            FillAppointmentStorage(schedulerStorage1.Appointments.Items, aptDataFileName);
        }

        static Stream GetFileStream(string fileName) {
            return new StreamReader(fileName).BaseStream;
        }

        static void FillResourceStorage(ResourceCollection c, string fileName) {
            using (Stream stream = GetFileStream(fileName)) {
                c.ReadXml(stream);
                stream.Close();
            }
        }
        static void FillAppointmentStorage(AppointmentCollection c, string fileName) {
            using (Stream stream = GetFileStream(fileName)) {
                c.ReadXml(stream);
                stream.Close();
            }
        }
        #endregion

        private void schedulerStorage_AppointmentsChanged(object sender, PersistentObjectsEventArgs e) {
            schedulerStorage1.Appointments.Items.WriteXml(aptDataFileName);
        }

        private void schedulerControl1_PopupMenuShowing(object sender, PopupMenuShowingEventArgs e)
        {
            // Check if it's the default menu of a Scheduler.
            if (e.Menu.Id == SchedulerMenuItemId.DefaultMenu)
            {

                // Disable the "New Recurring Appointment" menu item.
                e.Menu.DisableMenuItem(SchedulerMenuItemId.NewRecurringAppointment);

                // Hide the "New Recurring Event" menu item.
                e.Menu.RemoveMenuItem(SchedulerMenuItemId.NewRecurringEvent);

                // Enable the "Go To Today" menu item.
                e.Menu.EnableMenuItem(SchedulerMenuItemId.GotoToday);

                // Find the "New Appointment" menu item and rename it.
                SchedulerMenuItem item = e.Menu.GetMenuItemById(SchedulerMenuItemId.NewAppointment, false);
                if (item != null) item.Caption = "&Rent a Car";

                // Find the "New All Day Event" menu item and rename it.
                item = e.Menu.GetMenuItemById(SchedulerMenuItemId.NewAllDayEvent, false);
                if (item != null) item.Caption = "&Perform a Maintenance";

                SchedulerMenuItem myMenu = new SchedulerMenuItem("Click!", new EventHandler(MyMenuItemClick));
                myMenu.Image = imageList1.Images[0];
                e.Menu.Items.Add(myMenu);
            }
        }

        private void MyMenuItemClick(object sender, EventArgs e) {
            Appointment apt = schedulerStorage1.CreateAppointment(AppointmentType.Normal);
            apt.Start = schedulerControl1.SelectedInterval.Start;
            apt.End = schedulerControl1.SelectedInterval.End;
            apt.Subject = "Fly to Iceland!";
            schedulerStorage1.Appointments.Add(apt);
            System.Diagnostics.Process.Start("http://www.icelandtouristboard.com/");            
        }

        private void schedulerControl1_EditAppointmentFormShowing(object sender, AppointmentFormEventArgs e) {
            // Fill in the Subject field for the newly created appointment.
            if (((SchedulerControl)sender).Storage.Appointments.IsNewAppointment(e.Appointment)) {
               if (e.Appointment.AllDay) {
                   e.Appointment.Subject = "Perform a maintenance";
               }
               else {
                   e.Appointment.Subject="Rent a car";
               };
           }
       }
   }
}

File: VB/CustomContextMenu/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.IO
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Drawing
Imports DevExpress.Utils

Namespace CustomContextMenu
    Partial Public Class Form1
        Inherits Form

        Private Const aptDataFileName As String = "..\..\Data\appointments.xml"
        Private Const resDataFileName As String = "..\..\Data\resources.xml"

        Public Sub New()
            InitializeComponent()
            schedulerControl1.Start = New Date(2010, 7, 05)
            schedulerControl1.DayView.TopRowTime = New TimeSpan(16, 0, 0)
            FillData()
        End Sub

        #Region "FillData"
        Private Sub FillData()
            Dim customNameMapping As New AppointmentCustomFieldMapping("CustomName", "CustomName")
            Dim customStatusMapping As New AppointmentCustomFieldMapping("CustomStatus", "CustomStatus")
            schedulerStorage1.Appointments.CustomFieldMappings.Add(customNameMapping)
            schedulerStorage1.Appointments.CustomFieldMappings.Add(customStatusMapping)
            FillResourceStorage(schedulerStorage1.Resources.Items, resDataFileName)
            FillAppointmentStorage(schedulerStorage1.Appointments.Items, aptDataFileName)
        End Sub

        Private Shared Function GetFileStream(ByVal fileName As String) As Stream
            Return (New StreamReader(fileName)).BaseStream
        End Function

        Private Shared Sub FillResourceStorage(ByVal c As ResourceCollection, ByVal fileName As String)
            Using stream As Stream = GetFileStream(fileName)
                c.ReadXml(stream)
                stream.Close()
            End Using
        End Sub
        Private Shared Sub FillAppointmentStorage(ByVal c As AppointmentCollection, ByVal fileName As String)
            Using stream As Stream = GetFileStream(fileName)
                c.ReadXml(stream)
                stream.Close()
            End Using
        End Sub
        #End Region

        Private Sub schedulerStorage_AppointmentsChanged(ByVal sender As Object, ByVal e As PersistentObjectsEventArgs) Handles schedulerStorage1.AppointmentsChanged, schedulerStorage1.AppointmentsInserted, schedulerStorage1.AppointmentsDeleted
            schedulerStorage1.Appointments.Items.WriteXml(aptDataFileName)
        End Sub

        Private Sub schedulerControl1_PopupMenuShowing(ByVal sender As Object, ByVal e As PopupMenuShowingEventArgs) Handles schedulerControl1.PopupMenuShowing
            ' Check if it's the default menu of a Scheduler.
            If e.Menu.Id = SchedulerMenuItemId.DefaultMenu Then

                ' Disable the "New Recurring Appointment" menu item.
                e.Menu.DisableMenuItem(SchedulerMenuItemId.NewRecurringAppointment)

                ' Hide the "New Recurring Event" menu item.
                e.Menu.RemoveMenuItem(SchedulerMenuItemId.NewRecurringEvent)

                ' Enable the "Go To Today" menu item.
                e.Menu.EnableMenuItem(SchedulerMenuItemId.GotoToday)

                ' Find the "New Appointment" menu item and rename it.
                Dim item As SchedulerMenuItem = e.Menu.GetMenuItemById(SchedulerMenuItemId.NewAppointment, False)
                If item IsNot Nothing Then
                    item.Caption = "&Rent a Car"
                End If

                ' Find the "New All Day Event" menu item and rename it.
                item = e.Menu.GetMenuItemById(SchedulerMenuItemId.NewAllDayEvent, False)
                If item IsNot Nothing Then
                    item.Caption = "&Perform a Maintenance"
                End If

                Dim myMenu As New SchedulerMenuItem("Click!", New EventHandler(AddressOf MyMenuItemClick))
                myMenu.Image = imageList1.Images(0)
                e.Menu.Items.Add(myMenu)
            End If
        End Sub

        Private Sub MyMenuItemClick(ByVal sender As Object, ByVal e As EventArgs)
            Dim apt As Appointment = schedulerStorage1.CreateAppointment(AppointmentType.Normal)
            apt.Start = schedulerControl1.SelectedInterval.Start
            apt.End = schedulerControl1.SelectedInterval.End
            apt.Subject = "Fly to Iceland!"
            schedulerStorage1.Appointments.Add(apt)
            System.Diagnostics.Process.Start("http://www.icelandtouristboard.com/")
        End Sub

        Private Sub schedulerControl1_EditAppointmentFormShowing(ByVal sender As Object, ByVal e As AppointmentFormEventArgs) Handles schedulerControl1.EditAppointmentFormShowing
            ' Fill in the Subject field for the newly created appointment.
            If DirectCast(sender, SchedulerControl).Storage.Appointments.IsNewAppointment(e.Appointment) Then
               If e.Appointment.AllDay Then
                   e.Appointment.Subject = "Perform a maintenance"
               Else
                   e.Appointment.Subject="Rent a car"
               End If
            End If
        End Sub
    End Class
End Namespace